<a href="https://colab.research.google.com/github/Zauqyaraff/PZSIB-AML/blob/main/Topic9_10_HW_Zauqy_Affan_Raff.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#### Import package

In [1]:
import scipy.stats as stats
import pandas as pd
import numpy as np

Make a list of the number of transactions for 20 customers as listed in the pdf file

In [2]:
data = np.array([0, 150, 50, 100, 130, 120, 100, 110, 75, 65, 150, 120, 50, 100, 100, 140, 90, 150, 50, 90])
data

array([  0, 150,  50, 100, 130, 120, 100, 110,  75,  65, 150, 120,  50,
       100, 100, 140,  90, 150,  50,  90])

#### Define H0 and H1, follow the steps to do T-test, and conclude the result of the campaign testing

In [3]:
#H0 = The average sale after training = $100
#H1 = The average sale after training > $100

t_statistic, p_value = stats.ttest_1samp(a=data, popmean=100)
print(t_statistic , p_value)

-0.3361296649141257 0.7404567835122086


Statistic is 0.3361296649141257 and pvalue is 0.7404567835122086. Because the pvalue for the given problem is more than 0.05 or 5% which is the alpha value, we accept the H0 while H1 is rejected. Conclusion: The average sale after training is equal to $100, so the campaign was not effective.

In [4]:
#Two tailed test:
stats.t.ppf(1-(0.05/2), 19)

2.093024054408263

Critical Region: t < - 2.09 or t > 2.09. Because t statistic is not in the critical region, then we fail to reject H0. Conclusion: The average sale after training is equal to $100, so the campaign was not effective.

In [5]:
#One tailed test:
p_value/2

0.3702283917561043

P-value/2 larger than alpha, meaning we fail to reject H0. Conclusion: The average sale after training is equal to $100, so the campaign was not effective.

In [6]:
#Critical Region
# If H1: mean > 100 (right side)
stats.t.ppf(1-0.05, 19)

1.729132811521367

Critical Region: t > 1.73. Because t statistic is not in the critical region, then we fail to reject H0. Conclusion: The average sale after training is equal to $100, so the campaign was not effective.

In [7]:
# If H1: mean < 50 (left side)
stats.t.ppf(0.05, 19)

-1.7291328115213678

Critical Region: t < -1.73. Because t statistic is not in the critical region, then we fail to reject H0. Conclusion: The average sale after training is equal to $100, so the campaign was not effective.